Add service logic

senyamac opened the pull request.

senyamac committed:
Add new files

Macost commented on src/main/java/com/example/getnewjoke/kafka/KafkaJokeProducer.java:
```diff
@@ -0,0 +1,49 @@
+
+  public KafkaJokeProducer() {
+    props = new Properties();
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, AppConst.WSL_IP + ":9092");
```
port is also could be inserted as a constant
but more clear way to set it as a whole parameter and name it as `kafkaBootstrapUrl `for example

senyamac replied:
Done

Macost commented on src/main/java/com/example/getnewjoke/kafka/KafkaJokeProducer.java:
```diff
@@ -0,0 +1,49 @@
+    props = new Properties();
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, AppConst.WSL_IP + ":9092");
+    props.put(
+        AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, "http://" + AppConst.WSL_IP + ":8081");
```
the same comment as above, `schemaRegistryUrl`

Macost replied:
and please import all this configuration properties through the `@Value` spring annotation 
it should look like mentioning the field as follows:
```
@Value("${spring.kafka.properties.schema.registry.url}")
private String schemaRegistryUrl;
```
and the parameter value set in the application.yaml

senyamac replied:
Done

Macost commented on src/main/java/com/example/getnewjoke/kafka/KafkaJokeProducer.java:
```diff
@@ -0,0 +1,49 @@
+
+  public void addNewJokeToKafka(JokeEntity jokeEntity) {
+    final Joke joke = new Joke(String.valueOf(jokeEntity.getId()), jokeEntity.getCategory(), jokeEntity.getType(), jokeEntity.getJoke());
+    final ProducerRecord<String, Joke> record = new ProducerRecord<String, Joke>(AppConst.TOPIC, joke.getId().toString(), joke);
```
please apply code auto-formatting Ctrl+Alt+L
it will follow code style that you imported in IDEA

senyamac replied:
Done

Macost commented on src/main/java/com/example/getnewjoke/service/NewJokeService.java:
```diff
@@ -0,0 +1,55 @@
+  }
+
+  public JokeEntity getNewJokeFromApi() {
+    String message = restTemplate.getForObject(AppConst.URL_API, String.class);
```
move URL_API to application.yaml

senyamac replied:
Done

Macost commented on src/main/java/com/example/getnewjoke/service/NewJokeService.java:
```diff
@@ -0,0 +1,55 @@
+    ObjectMapper mapper = new ObjectMapper();
+    JokeEntity jokeEntity  = new JokeEntity();
+    try {
+      jokeEntity = mapper.readValue(message, JokeEntity.class);
```
mapper could be declared locally, no need to describe it in line 37

senyamac replied:
Done

Macost commented on src/main/java/com/example/getnewjoke/service/NewJokeService.java:
```diff
@@ -0,0 +1,55 @@
+      jokeEntity = mapper.readValue(message, JokeEntity.class);
+      log.debug(jokeEntity.toString());
+    } catch (JsonProcessingException e) {
+      log.error(e.getMessage(), e);
```
I want you to start the message by describing this error yourself,
e.g.: `log.error("Error while trying to convert API response to JokeEntity, {} {}", e.getMessage(), e);`

senyamac replied:
Done

Macost commented on src/main/java/com/example/getnewjoke/service/NewJokeService.java:
```diff
@@ -0,0 +1,55 @@
+  }
+
+  @Scheduled(fixedDelay = 3000)
+  public void processWithJoke() {
```
what does `processWithJoke ` means?
either correct this or just rename it to `scheduledTask()`

senyamac replied:
Done

Macost commented on src/main/java/com/example/getnewjoke/util/AppConst.java:
```diff
@@ -0,0 +1,9 @@
+  public static final String WSL_IP = "172.27.13.237";
+  public static final String TOPIC = "jokes";
+
+  private AppConst() {}
```
no need in this file at all, use application.yaml

senyamac replied:
Done

Macost reviewed (changes requested):
please resolve comments

senyamac committed:
Move constant properties from AppConst.java to application.yml

senyamac committed:
Import properties from application.yml
Fix code style
Fix IDEA warnings

senyamac committed:
Fix NewJokeService.java comments
-Move objectMapper to class variables
-Import API URL from application.yml
-Fix code style
-Rename scheduled method
-Change error text

senyamac committed:
Change API

senyamac committed:
Refactor Joke entity for using another API

senyamac committed:
Add Dockerfile

senyamac committed:
Add processing for null rest response
